src/main/java/uk/gov/hmcts/darts/datamanagement/service/impl/InboundToUnstructuredProcessorSingleElementImpl.java

```diff
@@ -0,0 +1,319 @@
+import static org.apache.commons.codec.digest.DigestUtils.md5;
+import static uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum.UNSTRUCTURED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.AWAITING_VERIFICATION;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.FAILURE;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.FAILURE_CHECKSUM_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.FAILURE_FILE_NOT_FOUND;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.FAILURE_FILE_SIZE_CHECK_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.FAILURE_FILE_TYPE_CHECK_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+import static uk.gov.hmcts.darts.datamanagement.service.impl.InboundToUnstructuredProcessorImpl.FAILURE_STATES_LIST;
+
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class InboundToUnstructuredProcessorSingleElementImpl implements InboundToUnstructuredProcessorSingleElement {
+
+    private static final int INITIAL_VERIFICATION_ATTEMPTS = 1;
+    private static final int INITIAL_TRANSFER_ATTEMPTS = 1;
+
+    private final DataManagementService dataManagementService;
+    private final DataManagementConfiguration dataManagementConfiguration;
+    private final UserAccountRepository userAccountRepository;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final TranscriptionConfigurationProperties transcriptionConfigurationProperties;
+    private final AudioConfigurationProperties audioConfigurationProperties;
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final MediaRepository mediaRepository;
+
+    @Override
+    @Transactional
+    public void processSingleElement(ExternalObjectDirectoryEntity inboundExternalObjectDirectory,
+                                     List<ExternalObjectDirectoryEntity> unstructuredStoredList,
+                                     List<ExternalObjectDirectoryEntity> unstructuredFailedList) {
+
+        ExternalObjectDirectoryEntity unstructuredExternalObjectDirectoryEntity =
+            getNewOrExistingInUnstructuredStoredOrFailed(inboundExternalObjectDirectory,
+                                                         unstructuredStoredList,
+                                                         unstructuredFailedList);
+        ObjectRecordStatusEntity unstructuredStatus = unstructuredExternalObjectDirectoryEntity.getStatus();
+        if (unstructuredStatus == null
+            || unstructuredStatus.getId().equals(STORED.getId())
+            || attemptsExceeded(unstructuredStatus, unstructuredExternalObjectDirectoryEntity)) {
+            log.info("Skipping transfer for EOD ID: {}", inboundExternalObjectDirectory.getId());
```

Comment:
i think this is going to be filling the logs every time it runs.

Reply:
changed to trace, cheers